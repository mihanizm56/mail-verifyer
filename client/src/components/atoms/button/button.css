.button {
    border: 1px solid black;
    border-radius: 15px;
    transition: .5s;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 300px;
    min-height: 70px;
    transition: .2s;
    outline: none;
}

.button:focus {
    box-shadow: 0 0 0 2pt grey;
}

.button:hover {
    background-color: var(--black);
    --color-button-text: #ffffff;
}

@media screen and (max-width: 767px) {}