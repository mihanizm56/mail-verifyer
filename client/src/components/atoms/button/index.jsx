import React from "react";
import { OpenSansText } from "../../../components";
import sn from "classnames";
import "./button.css";

export const LinkButton = ({ text, handleButtonClick }) => (
	<div className={sn("button")} onClick={handleButtonClick}>
		<OpenSansText text={text} isButton />
	</div>
);

LinkButton.defaultProps = {
	handleButtonClick: () => console.log("default handleButtonClick"),
};
