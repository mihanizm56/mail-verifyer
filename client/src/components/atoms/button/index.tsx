import React, { memo } from "react";
import { OpenSansText } from "../../../components";
import classNames from "classnames";
import "./button.css";

type ButtonProps = {
	text: string,
	handleButtonClick?: () => void,
	type: "button" | "submit" | "reset" | undefined,
};

export const Button = memo(({ text, handleButtonClick, type }: ButtonProps) => (
	<button type={type} className={classNames("button")} onClick={handleButtonClick}>
		<OpenSansText text={text} isButton />
	</button>
));
