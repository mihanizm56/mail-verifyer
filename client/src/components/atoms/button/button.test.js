import React,{createElement} from "react";
import { shallow } from "enzyme";
import { Button } from "../button";

jest.doMock("../open-sans-text", ()=>{
	const OpenSansText = (props) => createElement('OpenSansText', props);
	return {OpenSansText}
});

describe("test Button", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const props = {
				text: "test_text",
				handleButtonClick: jest.fn(),
				type: "submit",
			};
			const component = shallow(<Button {...props} />);

			expect(component).toMatchSnapshot();
		});
		xit("render OpenSansText", () => {
			const props = {
				text: "test_text",
				handleButtonClick: jest.fn(),
				type: "submit",
			};
			const component = shallow(<Button {...props} />);

			expect(component.find(OpenSansText).length).toBe(1);
		});
	});
});
