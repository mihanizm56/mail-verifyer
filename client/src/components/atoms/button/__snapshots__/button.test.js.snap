// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button test render contract render snap with all props 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
  type="submit"
>
  <OpenSansText
    isButton={true}
    text="test_text"
  />
</button>
`;
